dialogue-httpurlconnection-client/src/main/java/com/palantir/dialogue/httpurlconnection/HttpUrlConnectionBlockingChannel.java

```diff
@@ -0,0 +1,171 @@
+        this.config = config;
+        // Sanitize path syntax and strip all irrelevant URL components
+        Preconditions.checkArgument(
+                null == Strings.emptyToNull(baseUrl.getQuery()),
+                "baseUrl query must be empty",
+                UnsafeArg.of("query", baseUrl.getQuery()));
+        Preconditions.checkArgument(
+                null == Strings.emptyToNull(baseUrl.getRef()),
+                "baseUrl ref must be empty",
+                UnsafeArg.of("ref", baseUrl.getRef()));
+        Preconditions.checkArgument(
+                null == Strings.emptyToNull(baseUrl.getUserInfo()), "baseUrl user info must be empty");
+        this.baseUrl = UrlBuilder.from(baseUrl);
+    }
+
+    @Override
+    public Response execute(Endpoint endpoint, Request request) throws IOException {
+        // Create base request given the URL
+        UrlBuilder url = baseUrl.newBuilder();
+        endpoint.renderPath(request.pathParams(), url);
+        request.queryParams().forEach(url::queryParam);
+        URLConnection urlConnection = url.build().openConnection();
+        if (!(urlConnection instanceof HttpURLConnection)) {
+            throw new SafeIllegalStateException(
+                    "Expected an HttpUrlConnection", SafeArg.of("actual", urlConnection.getClass()));
+        }
+        HttpURLConnection connection = (HttpURLConnection) urlConnection;
+        connection.setRequestMethod(endpoint.httpMethod().name());
+
+        // Fill headers
+        for (Map.Entry<String, String> header : request.headerParams().entrySet()) {
+            connection.addRequestProperty(header.getKey(), header.getValue());
+        }
+
+        connection.setRequestProperty("accept-encoding", "gzip");
+
+        connection.setConnectTimeout(Ints.checkedCast(config.connectTimeout().toMillis()));
+        connection.setReadTimeout(Ints.checkedCast(config.readTimeout().toMillis()));
+
+        // match okhttp behavior
+        connection.setInstanceFollowRedirects(true);
+
+        // Never ask users for credentials
+        connection.setAllowUserInteraction(false);
+        connection.setDoOutput(request.body().isPresent());
+        connection.setDoInput(true);
+
+        if (connection instanceof HttpsURLConnection) {
+            HttpsURLConnection httpsConnection = (HttpsURLConnection) connection;
+            httpsConnection.setSSLSocketFactory(config.sslSocketFactory());
+            // TODO(ckozak): hostname verifier, otherwise the default is used.
+            // It's likely fine for most scenarios, but we want consistency
+            // across all channels.
+        }
+
+        if (request.body().isPresent()) {
+            if (endpoint.httpMethod() == HttpMethod.GET) {
+                throw new SafeIllegalArgumentException("GET endpoints must not have a request body");
+            }
+            if (endpoint.httpMethod() == HttpMethod.DELETE) {
```

Comment:
J9 client has this same limitation which is unfortunate given that Conjure does allow for DELETEs with bodies

Reply:
I don't think HttpURLConnection has this limitation, I added validation to avoid behavior differences between clients. Should we allow clients to send deletes with bodies?

Reply:
I think if possible we should support DELETE with bodies given that valid conjure definitions may include them

Reply:
The jdk9 client can support them if we ask it, I'll include that in this change.

Reply:
I had been seeing runtime failures when trying to get it to make delete requests with bodies

Reply:
The test makes a delete request with a body and passes :-)